xcm/xcm-simulator/example/src/parachain.rs

```diff
@@ -149,6 +149,62 @@ impl Config for XcmConfig {
 	type SubscriptionService = ();
 }
 
+mod parachain_2 {
+	use super::*;
+
+	#[derive(Encode, Decode)]
+	pub enum MockPalletCall {
+		// In order to construct the Call enum for parachain_2's runtime
+		// we need to know the index of the particular extrinsic we wish to call
+		// which happens to be 0.
+		#[codec(index = 0)]
+		MockExtrinsic,
+	}
+
+	#[derive(Encode, Decode)]
+	pub enum Parachain2Call {
+		// In order to construct the Call enum for parachain_2's runtime
+		// we need to know the index of the Pallet we wish to access
+		// which happens to be 42 as defined in parachain_2's construct runtime.
+		#[codec(index = 42)]
+		MockPallet(MockPalletCall),
+	}
+}
+
+#[frame_support::pallet]
+pub mod cross_parachain_pallet {
+	use super::*;
+	use parachain_2::*;
+	use frame_support::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+	}
+
+	#[pallet::event]
+	pub enum Event<T: Config> {}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {}
+
+	impl<T: Config> Pallet<T>
+	{
+		pub fn construct_parachain_2_extrinsic_call() -> Parachain2Call {
+			Parachain2Call::MockPallet(MockPalletCall::MockExtrinsic)
```

Comment:
This is absolutely not the right way to test it, and in fact I don't even think you need an extra parachain just to test `Transact`. What you need to do instead is create a dispatchable under `#[pallet::call]`, and use the `Call` enum from `construct_runtime` to build a `Call` type that corresponds to the `#[pallet::call]`.

Reply:
Ok Keith will do.

Reply:
> This is absolutely not the right way to test it, and in fact I don't even think you need an extra parachain just to test `Transact`. What you need to do instead is create a dispatchable under `#[pallet::call]`, and use the `Call` enum from `construct_runtime` to build a `Call` type that corresponds to the `#[pallet::call]`.

I believe the intent of the PR was to particularly test XCM _across_ two parachains as part of a testcase and the approach was somewhat inspired by how by acala or moonbeam parachains approach xcm (from memory, not sure which one, but I've seen the manual call type definitions for the other chain before.

Reply:
@KiChjang does all the feedback you gave still stand given @drahnr statement? Just checking. Much appreciated for all the time and consideration from both of you.